doc/python-type-hinting.md

```diff
@@ -40,18 +40,16 @@ def get_objects_matching(ids: Sequence[str]) -> dict[int, dict]
 
```

Comment:
There are references to "Optional" still

Reply:
Reference to optional refactored with pipe operator